pkg/clusteragent/admission/mutate/autoinstrumentation/namespace_mutator.go

```diff
@@ -304,9 +336,14 @@ func (m *mutatorCore) newInitContainerMutators(
 
 // newInjector creates an injector instance for this pod.
 func (m *mutatorCore) newInjector(pod *corev1.Pod, startTime time.Time, lopts libRequirementOptions) *injector {
+	imageTag := m.config.Instrumentation.InjectorImageTag
+	if m.tagResolver != nil {
```

Comment:
Two notes:

1. We should consider _always_ providing a `resolver` so that we don't have to do `nil` checks and we can create a `noop` and use an interface to always be able to use it
2. I think there's a bug here since it might be that the `InjectorImageTag` is just a tag like `"0"` and i'm not sure where the resolver is checking which _repo_ this image is for, only passing in the tag and the registry, we might be missing the repo name in the registry here?